test/TesterInternal/StreamingTests/SingleStreamTestRunner.cs

```diff
@@ -372,6 +374,54 @@ private void Heading(string testName)
             await StopProxies();
         }
 
+        public async Task StreamTest_22_TestImmutabilityDuringStreaming()
+        {
+            Heading("StreamTest_22_TestImmutabilityDuringStreaming");
+
+            IStreamingImmutabilityTestGrain itemProducer = GrainClient.GrainFactory.GetGrain<IStreamingImmutabilityTestGrain>(Guid.NewGuid());
+            string producerSilo = await itemProducer.GetSiloIdentifier();
+
+            // Setup consumers one in silo of item producer and the other one in another silo.
```

Comment:
I don't understand why silo matters?  If a producer and consumer are local and stream is immutable, the same object is transferred, whereas remotely a new object is created, but the difference between immutable calls is whether there is a data copy made at call time, in which silo location is not relevant.